src/tests/utils.py

```diff
@@ -21,33 +31,86 @@ def login(self, username, password):
         self.find('.submit.button').click()
 
 
+class CodalabDaphneProcess(DaphneProcess):
+
+    # have to set port in this hidden way so we can override it later (Daphne uses
+    # fancy multiprocessing shared memory vars we have to work around
+    _port = 36475
+
+    def __init__(self, host, application, kwargs=None, setup=None, teardown=None):
+        super().__init__(host, application, kwargs=None, setup=None, teardown=None)
+
+        # Move our port into shared memory
+        self.port.value = self._port
+
+    def run(self):
+        """Overriding this _just_ for the port bit...!"""
+        try:
+            # Create the server class -- with our fancy multiprocessing variable (note
+            # `self.port.value`)
+            endpoints = build_endpoint_description_strings(host=self.host, port=self.port.value)
+            self.server = Server(
+                application=self.application,
+                endpoints=endpoints,
+                signal_handlers=False,
+                **self.kwargs
+            )
+            # Set up a poller to look for the port
+            reactor.callLater(0.1, self.resolve_port)
+            # Run with setup/teardown
+            self.setup()
+            try:
+                self.server.run()
+            finally:
+                self.teardown()
+        except Exception as e:
+            # Put the error on our queue so the parent gets it
+            self.errors.put((e, traceback.format_exc()))
+
+
 @pytest.mark.e2e
-class SeleniumTestCase(CodalabTestHelpersMixin, StaticLiveServerTestCase):
+class SeleniumTestCase(CodalabTestHelpersMixin, ChannelsLiveServerTestCase):
     urls = 'urls'  # TODO: what the F is this???
     serialized_rollback = True
 
+    ProtocolServerProcess = CodalabDaphneProcess
     host = '0.0.0.0'
+    serve_static = True
 
-    test_files_dir = f'{os.getcwd()}/src/tests/functional/test_files'
+    # test_files_dir = f'{os.getcwd()}/src/tests/functional/test_files'
+    test_files_dir = f'/test_files'
 
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
         cls.host = socket.gethostbyname(socket.gethostname())
+        cls.circle_dir = os.environ.get('CIRCLE_ARTIFACTS', os.path.join(os.getcwd(), "artifacts/"))
+
+        # Setup console.log logging
+        desired_capabilities = DesiredCapabilities.FIREFOX
+        desired_capabilities['loggingPrefs'] = {'browser': 'ALL'}
+
         cls.selenium = webdriver.Remote(
             command_executor=f'http://{settings.SELENIUM_HOSTNAME}:4444/wd/hub',
-            desired_capabilities=DesiredCapabilities.FIREFOX,
+            desired_capabilities=desired_capabilities,
         )
-        # Wait 10 seconds for elements to appear, always
-        cls.selenium.implicitly_wait(10)
 
     @classmethod
     def tearDownClass(cls):
         cls.selenium.quit()
+
+        # TODO: Saving logs is not working, gets HTTP 405 not allowed. Maybe our `desired_capabilities` need to change
```

Comment:
Chrome is the only browser this will work in. For some reason firefox doesn't let you interact with logs. I've tried repeatedly to find answers on SO, and everybody's response is "yea, use Chrome if you want to save console logs". But then, I also couldn't find anything about VNCing in to circleCI, so maybe you'll get luckier than me.

Reply:
Could maybe leave this here for the next attempt, with that additional note re: Chrome?